servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -126,73 +144,94 @@ public void unlockBlock(long userId, long blockId) throws NotFoundException {
   @Override
   public BlockWriter getBlockWriter(long userId, long blockId) throws NotFoundException,
       IOException {
-    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
-    return new LocalFileBlockWriter(tempBlockMeta);
+    // NOTE: a temp block is supposed to be visible for its own writer, unnecessary to acquire
```

Comment:
Also, are we planning to handle multiple writers to the same block in this PR? (basically making the temp blocks keyed off user id and block id instead of just block id).